Dockerfile

```diff
@@ -1,14 +1,29 @@
-FROM golang:alpine
+# Build container
+FROM --platform=${BUILDPLATFORM} golang:1.23 AS builder
 LABEL org.opencontainers.image.source=https://github.com/devcyclehq/sdk-proxy
 LABEL org.opencontainers.image.description="DevCycle SDK Proxy"
 LABEL org.opencontainers.image.licenses=MIT
 
-ENV CGOENABLED=0
+ARG TARGETARCH
+ARG TARGETOS
+
+ENV CGO_ENABLED=0
 
 WORKDIR /usr/src/app
 
-COPY . .
-RUN go mod download
-RUN go build -v -o /usr/local/bin/devcycle-sdk-proxy ./cmd
+RUN --mount=type=bind \
+    --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target=/root/.cache/go-build \
+    GOARCH="$TARGETARCH" GOOS="$TARGETOS" \
+    go build -v -ldflags="-s -w" \
```

Comment:
I'd probably keep the symbols here for debugging; I don't particularly need to minimize this image down to the tiniest static binary possible; but it has proven helpful in the past to have symbols linked.